src/kernels/helpers.ts

```diff
@@ -1574,3 +1576,29 @@ export function deserializeKernelConnection(kernelConnection: any): KernelConnec
     }
     return kernelConnection;
 }
+
+export function createLiveLocalKernelConnectionMetadata(
+    kernel: IKernel
+): LiveLocalKernelConnectionMetadata | undefined {
+    if (kernel.kernelConnectionMetadata.kind === 'startUsingLocalKernelSpec') {
+        return {
+            id: `liveLocal:${kernel.kernelConnectionMetadata.id}`,
+            kernelId: kernel.id,
+            kernelSpec: kernel.kernelConnectionMetadata.kernelSpec,
+            kernelConnectionMetadataKind: kernel.kernelConnectionMetadata.kind,
+            kind: 'connectToLiveLocalKernel',
+            interpreter: kernel.kernelConnectionMetadata.interpreter,
+            kernelModel: undefined
+        };
+    } else if (kernel.kernelConnectionMetadata.kind === 'startUsingPythonInterpreter') {
+        return {
+            id: `liveLocal:${kernel.kernelConnectionMetadata.id}`,
+            kernelId: kernel.id,
+            kernelSpec: kernel.kernelConnectionMetadata.kernelSpec,
+            kernelConnectionMetadataKind: kernel.kernelConnectionMetadata.kind,
+            kind: 'connectToLiveLocalKernel',
+            interpreter: kernel.kernelConnectionMetadata.interpreter,
+            kernelModel: undefined
```

Comment:
Are live local kernels not the same as remote? Wondering why they don't have a kernel model.

Reply:
No live local kernels are not the same. With live local kernels, we just communicate with the IKernel, there's no kernel model (which only exists when using Jupyter REST api to get information about active sessions).

I have thought about adding a kernelModel for live local kernels as well, as we could have the information such as # of connections, last activity and the like (which we use when dealing with remote kernels).
However that's soemthign that can always be added later on.